File: java/NeedleInHaystack.java
import java.util.Arrays;

public class NeedleInHaystack
{
    public static String findNeedle(Object[] haystack) {
        return "found the needle at position " + Arrays.asList(haystack).indexOf("needle");
    }
    public static void main(String[] args) {
		System.out.println(findNeedle(new Object[] {"hay", "junk", "hay", "hay", "moreJunk", "needle", "randomJunk"}));
	}
}

File: c#/NeedleInHayStack.cs
// return the index of where "needle" is found in an array

using System;

public class NeedleInHayStack
{

	public static string FindNeedle(object[] haystack)
	{
		return "found the needle at position " + Array.IndexOf(haystack, "needle");
	}

	public static void Main(string[] args)
	{
		Console.WriteLine(FindNeedle(new object[]{'3', "123124234", null, "needle", "world"})); // found the needle at position 3 
	}
}